sqlobject/src/main/java/org/jdbi/v3/sqlobject/statement/internal/ResultReturner.java

```diff
@@ -430,4 +436,21 @@ protected void accept(Stream<?> stream, @SuppressWarnings("rawtypes") Consumer c
             }
         }
     }
+
+    static class ConsumeIterableResultReturner extends ConsumerResultReturner {
+
+        ConsumeIterableResultReturner(int consumerIndex, QualifiedType<?> elementType) {
+            super(consumerIndex, elementType);
+        }
+
+        @SuppressWarnings("unchecked")
+        @Override
+        protected void accept(Stream<?> stream, @SuppressWarnings("rawtypes") Consumer consumer) {
+            try {
+                consumer.accept(stream.collect(Collectors.toCollection(LinkedList::new)));
```

Comment:
This looks like it eagerly consumes all results. We should probably stream the results out through the returned iterator, if possible?